chall04/vde-dios.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+
+import sys
+import re
+
+
+def error(e , i):
+    file_name =  "stdin" if i == 0 else sys.argv[i]
+    if (e == 1):
+        print("%s: %s: Bad format" %(sys.argv[0], file_name))
+    elif (e == 2):
+        print("%s: %s: Can't read file" %(sys.argv[0], file_name))
+    elif (e == 3):
+        print("%s: %s: Not enough space in the given shelves" %(sys.argv[0], file_name))
+
+
+# Check format plus sort shelves
+def shelf_parse(shelves):
+    l = shelves.replace("\n", "").split(" ")
+    l.sort(key=len, reverse=True)
+    for c in l:
+        if not c.isdigit():
+            return None
+    return (l)
+
+
+# Check book format and add books
+def book_parse(books):
+    b = re.search('^\d+', books)
+    if re.search('^\d+\s\w+', books) is None:
+        return None
+    return int(b.group())
```

Comment:
The reason for only using one space in **_re.search('^\d+\s\w+', books)_** was because in the subject was explicitly defined that book lines were have to be formed of a number + **a** space + tittle. 

About the usage of regexp is really smart, take note :)